src/main/java/com/zendesk/maxwell/bootstrap/SynchronousBootstrapper.java

```diff
@@ -182,8 +189,8 @@ private ResultSet getAllRows(String databaseName, String tableName, Schema schem
 	}
 
 	private Statement createBatchStatement(Connection connection) throws SQLException, InterruptedException {
-		Statement statement = connection.createStatement();
-		statement.setFetchSize(context.getConfig().bootstrapperBatchFetchSize);
+		Statement statement = connection.createStatement(ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
+		statement.setFetchSize(Integer.MIN_VALUE);
```

Comment:
So I went and looked into this further; according to http://stackoverflow.com/questions/26046234/is-there-a-sql-jdbc-that-will-respect-fetchsize/36986532#36986532, with `useCursorFetch=true` we were supposed to not buffer the whole damn thing into memory.  You're definitely right about using making mistakes writing to the connection while we're still reading results, but lemme poke a bit more at this tomorrow.

But also just turning this into a streaming query is likely totally fine.